velox/exec/MergeJoin.h

```diff
@@ -20,6 +20,28 @@
 #include "velox/exec/Operator.h"
 
 namespace facebook::velox::exec {
+
+/// The merge join operator assumes both streams, left (from addInput()) and
+/// right (from rightSource), are sorted in the join key.
+///
+/// It works by identifying and maintaining a window of rows with key matches
+/// (leftMatch_ and rightMatch_), and outputting a cartesian product of each key
+/// match. Since keys can span multiple vectors, multiple batches from either
+/// side may need to be materialized and kept in memory. Therefore, the memory
+/// requirement is proportional to the size of the longest key match. Once all
+/// output for a particular key match is produced, the respective batches are
+/// discarded.
+///
+/// Output is produced outputBatchSize_ rows at a time.
+///
+/// The merge join operator generally returns dictionaries which are wrapped
+/// around input vectors. The output is aligned to left vectors, and since
+/// dictionaries cannot wrap around more than one vector, at times merge join
+/// may return fewer than outputBatchSize_ rows.
+///
+/// Dictionaries for right projections are generated optimistically created; we
```

Comment:
> /// Dictionaries for right projections are generated optimistically created;

typos: "are generated ... created" - one of 'generated' or 'created' may need to be removed